### PR TITLE
Initial add post and get 

### PR DESCRIPTION
**Pull request by SoumendraPanda**

### CODE REVIEW

File: backend/src/main/java/com/urlshortner/backend/controller/Controller.java [hunk 1]
```diff
@@ -1,15 +1,76 @@
...
+    @PostMapping("/shorten")
+    @ResponseBody
+    public String shortenUrl(@RequestBody String originalUrl) {
+        logger.info("Received request to shorten URL: {}", originalUrl);
+
+        // Extract the path part of the URL (not including the domain)
+        String path = extractPath(originalUrl);
+
+        // Generate a short key based on the extracted path
+        String shortKey = generateShortKey(path);
```
Comment (srinath-tripathy) [inline backend/src/main/java/com/urlshortner/backend/controller/Controller.java:35]: There can be same short key value in for multiple original Urls and there is a chance of getting 2 Original Urls for the same short key.

File: backend/src/main/java/com/urlshortner/backend/entity/UrlMapping.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+@Entity
+public class UrlMapping {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String originalUrl;
+
+    private String shortUrl;
```
Comment (srinath-tripathy) [inline backend/src/main/java/com/urlshortner/backend/entity/UrlMapping.java:16]: can we make this as a unique constraint?

File: backend/src/main/java/com/urlshortner/backend/controller/Controller.java [hunk 3]
```diff
@@ -1,15 +1,76 @@
...
     public String getHello(){
-        log.info("hello");
+       // log.info("hello");
         return "hello";
     }
+
+    @Autowired
+    private UrlMappingService service;
+
+    private static final Logger logger = LoggerFactory.getLogger(Controller.class);
```
Comment (srinath-tripathy) [inline backend/src/main/java/com/urlshortner/backend/controller/Controller.java:23]: This is not needed. Already using @slf4j annotation

**Comment by srinath-tripathy** [comment]
Merging it now . The commented things can be taken in later period of time. Nice Work :)

**Review by skt17 — APPROVED** [review]
Merging it now . The commented things can be taken in later period of time. Nice Work :)

**Pull request merged**